frontend/src/tests/UsernameInput.spec.ts

```diff
@@ -0,0 +1,171 @@
+import { it, describe, beforeEach, afterEach, expect, vi } from 'vitest';
+import { flushPromises, mount } from '@vue/test-utils';
+import PrimeVue from 'primevue/config';
+import { UserType } from 'fam-app-acsctl-api';
+import { AppActlApiService } from '@/services/ApiServiceFactory';
+import UserNameInput from '@/components/grantaccess/form/UserNameInput.vue';
+import { isLoading, setLoadingState } from '@/store/LoadingState';
+import type { VueWrapper } from '@vue/test-utils/dist/vueWrapper';
+import type { DOMWrapper } from '@vue/test-utils/dist/domWrapper';
+import type { AxiosResponse } from 'axios';
+import { fixPrimevueCssError } from '@/tests/common/fixPrimevueCssErr';
+
+fixPrimevueCssError()
+
+
+const userInputMock = (): AxiosResponse => {
+    return {
+        data: {
+            firstName: "Name",
+            found: true,
+            lastName: "LastName",
+            userId: 'userId'
+        },
+        status: 200,
+        statusText: 'Ok',
+        headers: {},
+        config: {},
+    }
+
+}
+
+describe('UserNameInput', () => {
+    let wrapper: VueWrapper;
+    let emitChange: unknown[][] | undefined;
+    let emitSetVerifyResult: unknown[][] | undefined;
+
+    let usernameInputText: DOMWrapper<HTMLElement>;
+    let usernameInputTextEl: HTMLInputElement;
+    let verifyButton: DOMWrapper<HTMLElement>;
+    let verifyButtonEl: HTMLButtonElement;
+
+    let cardUsernameEl: HTMLSpanElement;
```

Comment:
we don't need to declare all variables at this level, if the variable is only used by a single test case, it can be declared inside the test case.

Reply:
I'll see which ones are in only one test and refactor